### DISCUSSION TITLE
Pushing mark within `meow-goto-line`

### DISCUSSION THREAD

**Discussion by okamsn** [meow-edit/meow#397] [Ideas]
The command `goto-line`, if no region is active, sets the mark at the current position before jumping, so that it is easy to jump back to that position via `C-u C-SPC`. `meow-goto-line`'s use of `save-mark-and-excursion` when getting the chosen line's position prevents this from working.

Would you consider pushing the mark in `meow-goto-line` itself after getting the chosen line's position?

**Comment by DogLooksGood** [comment]
Well, it's hard. Because `meow-goto-line` will call the function assigned to `meow-goto-line-function`.

**Reply by okamsn** [reply]
I'm not sure what you mean. Which part is the hard part that you refer to?

For example, the below code is what I am testing right now. The pushing of the mark occurs after the calling of the function stored in `meow-goto-line-function`.

```elisp
  (defcustom my-meow-goto-line-push-mark t
    "Whether to push mark in `meow-goto-line'.")
  (defun my-meow-goto-line ()
    "Goto line, recenter and select that line.

This command will expand line selection."
    (interactive)
    (let* ((rbeg (when (use-region-p) (region-beginning)))
           (rend (when (use-region-p) (region-end)))
           (expand (equal '(expand . line) (meow--selection-type)))
           (orig-p (point))
           (beg-end (save-mark-and-excursion
                      (if meow-goto-line-function
                          (call-interactively meow-goto-line-function)
                        (meow--execute-kbd-macro meow--kbd-goto-line))
                      (cons (line-beginning-position)
                            (line-end-position))))
           (beg (car beg-end))
           (end (cdr beg-end)))
      ;; New expression:
      (when (and my-meow-goto-line-push-mark (not (use-region-p)))
        (push-mark orig-p))
      (thread-first
        (meow--make-selection '(expand . line)
                              (if (and expand rbeg) (min rbeg beg) beg)
                              (if (and expand rend) (max rend end) end))
        (meow--select (> orig-p beg)))
      (recenter)))
```

**Reply by DogLooksGood** [reply]
The `meow-goto-line-function` could be assigned to another function, not necessarily to be `goto-line`. By default, it's searched by keybinding.
